mixer/pkg/runtime2/config/kinds.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"github.com/gogo/protobuf/proto"
+
+	"istio.io/istio/mixer/pkg/adapter"
+	configpb "istio.io/istio/mixer/pkg/config/proto"
+	"istio.io/istio/mixer/pkg/template"
+	"istio.io/istio/pkg/log"
+)
+
+// KindMap generates a map from object kind to its proto message.
+func KindMap(adapterInfo map[string]*adapter.Info, templateInfo map[string]*template.Info) map[string]proto.Message {
```

Comment:
Is this new code or old code?

Reply:
Old. 
BTW https://github.com/istio/istio/pull/2450 captures bulk of the unmodified (or slightly modified) old code transfer.